src/controllers/user.js

```diff
@@ -34,7 +35,47 @@ const getSingleUser = async (req, res) => {
   }
 };
 
+// eslint-disable-next-line consistent-return
+const updateUser = async (req, res) => {
+  const bodyParams = Object.keys(req.body); // ['name', 'email', 'password']
+  const restrictedParams = [
+    '_id',
+    'email',
```

Comment:
- We might re-allow users to update their emails, since we're not going to do the restrictions anymore.
- I tested the endpoint on Postman, and the outcome isn't what we really expect:
![Screenshot 2022-08-15 155701](https://user-images.githubusercontent.com/102853969/184640399-64cc84ed-f580-46d7-81cd-976de4ae5f53.png)
![Screenshot 2022-08-15 155139](https://user-images.githubusercontent.com/102853969/184640428-b8421be8-20e8-4c30-8cd4-873f1b83c27a.png)
Those fields, address and username, will always be there, but when a user leaves them empty, because s/he doesn't want to update them, s/he will receive those errors. Plus, I don't think it should say `passwordConfirmation Cannot be empty` on img1, maybe `Password fields can not be empty`.
- Documentation looks fantastic.
- Endpoint name shouldn't include an id. It can be profile, and each `patch`, like username, can be like `/profile/username` or, preferably, `/profile/update/username`. This way, the GET of `/profile/update` will include those fields, and a user would choose what to update. Plus, we wouldn't need, then, the `restrictedParams`.
- We can skip updating addresses in this endpoint: address has its own model.

Reply:
- Now updating password is it's own endpoint
- endpoint now doesn't include an id